src/test/java/org/klodnicki/service/LendBookServiceTest.java

```diff
@@ -0,0 +1,226 @@
+            //given
+            //takes from beforeEach method
+            int expected = 99;
+
+            //when
+            lendBookService.lend("first name", "last name", "pesel", "title", "author",
+                    "edition");
+            //this is other object after lending it, so I need to pull it from repo to get updated copies number
+            BookInfo bookInfo2 = bookService.findBookByTitleAndAuthorAndEdition("title", "author",
+                    "edition");
+            int actual = bookInfo2.getCopiesNumber();
+
+            //then
+            assertEquals(expected, actual);
+        }
+
+        @Test
+        public void lend_ShouldAddLendingInformationToLendingRepo_WhenAllArgumentsAreCorrectAndPassAllValidations()
+                throws MaximumBookBorrowedLimitException, NotEnoughBookCopiesException, NotFoundInDatabaseException {
+            //given
+            //takes from beforeEach method
+            Long expected = lendingInformationRepository.count() + 1;
+
+            //when
+            lendBookService.lend("first name", "last name", "pesel", "title", "author",
+                    "edition");
+            Long actual = lendingInformationRepository.count();
+
+            //then
+            assertEquals(expected, actual);
+        }
+
+        @Test
+        public void hasMoreThanOneEdition_ShouldReturnTrue_WhenBookInfoOfTheSameTitleAndAuthorHasMoreThanOneEdition() {
+            //given
+            //takes from beforeEach method
+            BookInfo bookInfo2 = new BookInfo("title", "author", "isbn", "publisher", 2022,
+                    "edition2", "genre", "description", "language", 100);
+
+            bookRepository.add(bookInfo2);
+            boolean expected = true;
+
+            //when
+            //title and author must have the same values for bookInfo and bookInfo2
+
+            //then
+            assertEquals(expected, lendBookService.hasMoreThanOneEdition("title", "author"));
+        }
+    }
+
+
+    @Test
+    public void lend_ShouldThrowNotEnoughBookCopiesException_WhenAvailableBookCopiesIsLessThanZero() {
+        //given
+        BookInfo bookInfo = new BookInfo("title", "author", "isbn", "publisher", 2020,
+                "edition", "genre", "description",
+                "language", 0); // copies number must be less or equal to 0
+        Account account = new Account("first name", "second name", "last name", "pesel",
+                "phone number", "email", "address");
+
+        bookRepository.add(bookInfo);
+        accountRepository.create(account);
+
+        //when
+        assertThrows(NotEnoughBookCopiesException.class, () -> lendBookService.lend(account.getFirstName(), account.getLastName(),
+                account.getPesel(), bookInfo.getTitle(), bookInfo.getAuthor(), bookInfo.getEdition()));
+    }
+
+    @Test
+    public void lend_ShouldThrowMaximumBookBorrowedLimitException_WhenAccountBorrowedMaximumAmountOfBooks() {
```

Comment:
This code could be improved by using loops and collections. Please, take another look at it!

Reply:
perfect idea! DONE!